src/other/index.jl

```diff
@@ -345,22 +344,29 @@ Base.@propagate_inbounds parentcols(ind::SubIndex, ::Colon) = ind.cols
 
 Base.@propagate_inbounds parentcols(ind::SubIndex, idx::Not) = parentcols(ind, ind[idx])
 
-Base.@propagate_inbounds function SubIndex(parent::AbstractIndex, cols::AbstractUnitRange{Int})
+function SubIndex(parent::AbstractIndex, cols::AbstractUnitRange{Int})
     l = last(cols)
     f = first(cols)
-    @boundscheck if !checkindex(Bool, Base.OneTo(length(parent)), cols)
```

Comment:
Why not still allow using `@inbounds` to skip the checks? Doesn't seem it would harm.

Reply:
I have checked that the benefit is so small and in only a particular case of range that I thought it is not worth allowing it to make sure we are always consistent (as noted - if someone really wants speed one can call `SubIndex` directly - probably this is not advisable, but possible). This check is really cheap - most of the time compiler will optimize it out due to constant propagation.

But I can reintroduce it if you prefer.

Reply:
OK. So the `@boundscheck` is still used in places where it does make a difference?

Reply:
I have removed all `@boundscheck` in constructors (there were two places of this - this one and the one that now replaces `lazyremap!` functionality). But we do `@boundscheck` in other functions (that get called on the created objects).

Reply:
I thought at https://github.com/JuliaData/DataFrames.jl/pull/1958#discussion_r348938463 you said the performance impact was large for `DataFrameRow`?